tests/baselines/reference/override2.errors.txt

```diff
@@ -0,0 +1,32 @@
+tests/cases/conformance/override/override2.ts(12,14): error TS4113: This member must have an 'override' modifier because it overrides a member in the base class 'AB'.
+tests/cases/conformance/override/override2.ts(17,14): error TS4113: This member must have an 'override' modifier because it overrides a member in the base class 'AB'.
+
+
+==== tests/cases/conformance/override/override2.ts (2 errors) ====
+    abstract class AB {
+        abstract foo(v: string): void;
+    
+        abstract bar(v: string): void
+    }
+    
+    abstract class AD1 extends AB {
+    
+    }
+    
+    abstract class AD2 extends AB {
+        abstract foo(v: ''): void // need override?
+                 ~~~
+!!! error TS4113: This member must have an 'override' modifier because it overrides a member in the base class 'AB'.
+    }
+    
+    abstract class AD3 extends AB {
+        override foo(v: ''): void { } // need override?

```

Comment:
Oh, sorry if I missed earlier discussion on this, but I don’t think `override` should be required to implement an abstract method. (It’s not _really_ overriding anything.)

Reply:
Should we raise an error if use override with implement abstract method?

Reply:
That's what would make sense to me, but I don’t feel too strongly about it—@DanielRosenwasser what do you think?

Reply:
I guess if you rename in the base class, you won't run into issues. Still seems surprising.

One thing to be sure of is that if you are re-declaring an `abstract` member in a class, you MUST use an `override` keyword.

Reply:
Yeah, this is what I’d expect:

```ts
abstract class Base {
  abstract foo(): void;
  abstract bar(): void;
}

// No errors:
class Sub extends Base {
  override abstract foo(): number;
  bar() {
    return;
  }
}
```

Reply:
Peanut gallery comment:

I think `override` is useful to indicate that this method must match a parent type's declaration, so that you get an early error if your implementation doesn't match.

E.g. if you have `abstract class Parent { method(): string; }` and `class Child extends Parent { method() { return 'x'; } }`. Now `Parent` changes to `abstract class Parent { myMethod(): string; }`. You'll get an error about incorrectly extending `Parent` (OK), but it'd also be nice to get a message about `method()` no longer overriding (well, implementing) a parent signature.

This is particularly useful in large classes, where it might be confusing what changed and what method no longer matches.

Reply:
@andrewbranch I switched your comment to use `unknown` just because `void` in base, `number` in derived seems suspicious

Reply:
Yeah, that’s less sketchy, but `void` does work 🤷 https://www.typescriptlang.org/play?#code/IYIwzgLgTsDGEAJYBthjAgQmgpgg3gFAIKiQzwIBmA9jQBQCUAXAgG40CWAJgNzGlw0OIhDAoTVhx78AvoUJlhlFGgwBlAK4gEOAB4QcAO24ZsYPERJKKiWgxYIjmgLYgcUOUA

Reply:
@mprobst I think that the current message there is

```
Non-abstract class 'Child' does not implement inherited abstract member 'myMethod' from class 'Parent'.
```

is there something I'm missing?

Reply:
If users were able to put `override` on method, you'd get:

```
Non-abstract class 'Child' does not implement inherited abstract member 'myMethod' from class 'Parent'.
Method 'method' of 'Child' must override a super type method (or some such)
```

I'm arguing that it's useful to get the second error message, to more quickly determine what went wrong.

There's also the scenario where you refactor such that an interface you implement no longer needs you to implement `method`. With `override` on implementing methods, you'll get an error for code that's no longer useful.

Reply:
@mprobst seems like something somewhat similar to weak type checks - but what would that specific check be? When do you know that *some* method needs to be overridden? When everything is protected or private?